ledger/ledger-api-client/src/test/suite/scala/com/digitalasset/ledger/client/services/commands/CommandTrackerFlowTest.scala

```diff
@@ -453,27 +475,79 @@ class CommandTrackerFlowTest
       }
     }
 
+    "a completion without submission id arrives" should {
+      "fail if there are multiple pending commands with the same command id" in {
+        val Handle(submissions, _, unhandledF, completionStreamMock) =
+          runCommandTrackingFlow(allSubmissionsSuccessful)
+
+        submissions.sendNext(newSubmission("submissionId", "commandId"))
+        submissions.sendNext(newSubmission("anotherSubmissionId", "commandId"))
+
+        val completionWithoutSubmissionId =
+          Completion(
+            commandId,
+            Some(successStatus),
+            submissionId = "",
+          )
+        completionStreamMock.send(
+          CompletionStreamElement.CompletionElement(completionWithoutSubmissionId)
+        )
+
+        whenReady(unhandledF) { unhandled =>
+          unhandled should have size 2
+          unhandled should contain(
+            TrackedCommandKey("submissionId", "commandId") -> submission.context
+          )
+          unhandled should contain(
+            TrackedCommandKey("anotherSubmissionId", "commandId") -> submission.context
+          )
+        }
+      }
+
+      "output the completion" in {
```

Comment:
```suggestion
      "include the completion in the flow output" in {
```

Reply:
This name is consistent with other tests, so I think I'll leave it.